src/platforms/esp32/components/avm_builtins/i2c_driver.c

```diff
@@ -99,9 +106,36 @@ Context *i2c_driver_create_port(GlobalContext *global, term opts)
     struct I2CData *i2c_data = calloc(1, sizeof(struct I2CData));
     i2c_data->transmitting_pid = term_invalid_term();
 
-    term scl_io_num_term = interop_proplist_get_value(opts, SCL_IO_NUM_ATOM);
-    term sda_io_num_term = interop_proplist_get_value(opts, SDA_IO_NUM_ATOM);
-    term clock_hz_term = interop_proplist_get_value(opts, I2C_CLOCK_HZ_ATOM);
+    term scl_io_num_term = interop_kv_get_value_default(
+        opts, ATOM_STR("\xA", "scl_io_num"), term_invalid_term(), global
+    );
+    VALIDATE_NOT_INVALID(scl_io_num);
+
+    term sda_io_num_term = interop_kv_get_value_default(
+        opts, ATOM_STR("\xA", "sda_io_num"), term_invalid_term(), global
+    );
+    VALIDATE_NOT_INVALID(sda_io_num);
+
+    term clock_hz_term = interop_kv_get_value_default(opts,
+        ATOM_STR("\xC", "i2c_clock_hz"), term_invalid_term(), global
+    );
+    VALIDATE_NOT_INVALID(clock_hz);
+
+    i2c_data->i2c_num = I2C_NUM_0;
+    term i2c_num_term = interop_kv_get_value_default(
+        opts, ATOM_STR("\x7", "i2c_num"), term_invalid_term(), global
+    );
```

Comment:
Hm, but the `i2c_num` parameter is optional and defaults to `0`.  This is both for backwards compatibility, but also because it has a reasonable default and shouldn't need to be used in the majority of cases.

Unless I am missing something?

Reply:
got it, now it is very clear to me.